analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -52,13 +52,19 @@ operation.Instance.ConstantValue.Value switch
                 // Update DefaultValue when adding new types
                 true => SymbolicValue.True,
                 false => SymbolicValue.False,
-                null when (operation.Instance.Type ?? ConvertedType(operation.Parent)).CanBeNull() => SymbolicValue.Null,  // ToDo: MMF-2401, this will need to be reviewed
+                null when CanBeNull(operation) => SymbolicValue.Null,
                 string => SymbolicValue.NotNull,
                 _ => null
             };
 
         private static ITypeSymbol ConvertedType(IOperation operation) =>
             // Some version of Roslyn can send null here with "return default;". It's not reproducible by UTs, as we have "Type" set in that case.
             operation?.Kind == OperationKindEx.Conversion ? operation.ToConversion().Type : null;
+
+        private static bool CanBeNull(IOperationWrapperSonar operation)
+        {
+            var type = operation.Instance.Type ?? ConvertedType(operation.Parent);
+            return type is null || type.CanBeNull();
```

Comment:
Education: Can you explain to me what the case `type is null` is for? Is this some dynamic stuff? It is under test, but what is the test case?

Reply:
See the bottom part of PR description :)